cocos/renderer/scene/camera.js

```diff
@@ -450,17 +450,10 @@ export default class Camera {
     let cw = this._rect.w * width;
     let ch = this._rect.h * height;
 
-    // calculate w
-    let w =
-      worldPos.x * _matViewProj.m03 +
-      worldPos.y * _matViewProj.m07 +
-      worldPos.z * _matViewProj.m11 +
-      _matViewProj.m15;
-
     vec3.transformMat4(out, worldPos, _matViewProj);
-    out.x = cx + (out.x / w + 1) * 0.5 * cw;
-    out.y = cy + (out.y / w + 1) * 0.5 * ch;
-    out.z = out.z / w * 0.5 + 0.5;
+    out.x = cx + (out.x + 1) * 0.5 * cw;
```

Comment:
is it safe to remove division by w if projection matrix is used to multiplication?

Reply:
it's already done in `vec3.transformMat4`

Reply:
that is fine, thanks